scripts/ai/turns/TurnContext.lua

```diff
@@ -458,7 +458,10 @@ function TurnContext:getTurnEndNodeAndOffsets(steeringLength)
         -- implement in front of vehicle. Turn should end with the implement at the work start position, this is where
         -- the vehicle's root node is on the vehicleAtTurnEndNode
         turnEndNode = self.vehicleAtTurnEndNode
-        goalOffset = 0
+
+        -- set the turning endpoint a bit futher from the field starting point
+        -- for better aligment for front implements (e.g. potatoharvester)
+        goalOffset = -self.frontMarkerDistance
```

Comment:
Why the frontMarkerDistance? The turnEndNode is already frontMarkerDistance meters back from the work start. I agree some vehicles may need to finish the turn a bit further back but:
1. even with the potato harvester, front marker distance is 8-10 m or so, so this makes every turn 8-10 m bigger, causing problems when there are obstacles around the field
2. we have a similar problem with towed implements, and the issue is the same as the TODO further down in the code suggests (_TODO: this isn't exact science here, as the distance we need to straighten out the implement is rather a function of the radius, the starting angle and probably the tow bar length._) The extra distance we need is certainly not simply the front marker distance.

So here we'd need something that does not automatically increase the size of the turn for every vehicle and for those it increases, it does only as much as needed. 

I was too lazy to figure out the geometry for this, probably that's why we left it as is. I think, besides the front marker distance, a calculation should take into account the working width, the turning radius and probably even the depth of the work area.

Reply:
> Why the frontMarkerDistance?
It seemed like a good place to start. And yes, it makes every turn bigger and in case of potato harvester, a lot bigger. I made test capping the value to -4.0 meters which makes it less bigger, but still aligning good enough not to miss any crops.

For a very big machine like the potato harvester it is a fact it needs a lot of space to turn and without reversing enable or making headlands, it does extend quite far which I think is up to the user to set up the field/course properly.

This does make it much better for fields which does have enough space around them.

Reply:
With turn on field on + lower ealry it doesn't miss a single fruit already.